auth/token-managers/iam-token-manager.ts

```diff
@@ -37,38 +36,35 @@ function onlyOne(a: any, b: any): boolean {
 
 const CLIENT_ID_SECRET_WARNING = 'Warning: Client ID and Secret must BOTH be given, or the header will not be included.';
 
+/** Configuration options for IAM token retrieval. */
 interface Options extends TokenManagerOptions {
   apikey: string;
   clientId?: string;
   clientSecret?: string;
 }
 
-// this interface is a representation of the response
-// object from the IAM service, hence the snake_case
-// parameter names
-export interface IamTokenData {
-  access_token: string;
-  refresh_token: string;
-  token_type: string;
-  expires_in: number;
-  expiration: number;
-}
-
+/**
+ * The IAMTokenManager takes an api key and performs the necessary interactions with
+ * the IAM token service to obtain and store a suitable bearer token. Additionally, the IAMTokenManager
+ * will retrieve bearer tokens via basic auth using a supplied `clientId` and `clientSecret` pair.
+ */
 export class IamTokenManager extends JwtTokenManager {
   protected requiredOptions = ['apikey'];
   private apikey: string;
   private clientId: string;
   private clientSecret: string;
 
   /**
-   * IAM Token Manager Service
    *
-   * Retreives and stores IAM access tokens.
+   * Create a new [[IamTokenManager]] instance.
    *
-   * @param {Object} options
-   * @param {String} options.apikey
-   * @param {String} options.iamAccessToken
-   * @param {String} options.iamUrl - url of the iam api to retrieve tokens from
+   * @param {object} options Configuration options.
+   * @param {string} options.apikey The IAM api key.
+   * @param {string=} options.clientId The `clientId` and `clientSecret` fields are used to form a "basic"
+   *   authorization header for IAM token requests.
+   * @param {string=} options.clientSecret The `clientId` and `clientSecret` fields are used to form a "basic"
+   *   authorization header for IAM token requests.
+   * @param {string} [url='https://iam.cloud.ibm.com/identity/token'] The IAM endpoint for token requests.
```

Comment:
Just realized I hadn't been looking at this at all 🤦‍♂ but doesn't JSDoc use the pattern of wrapping optional parameters in square brackets like this `url` parameter? Do you think we should address this everywhere, like for `clientId` and `clientSecret`?

Reply:
I forgot about this as well.

Reply:
I think it is simple to do though.